### ISSUE TITLE
NewType pseudo-class not assignable to type[T] parameter

### ISSUE THREAD

**Issue by matthiasgoergens** [astral-sh/ty#2801]
Passing a `NewType` as an argument to a function expecting `type[T]` produces `invalid-argument-type`. Both mypy and pyright accept this pattern.

## Reproducer

```python
from typing import NewType, TypeVar

UserId = NewType("UserId", int)

T = TypeVar("T")

def make_id(cls: type[T], value: int) -> T:
    return cls(value)

x = make_id(UserId, 42)  # error[invalid-argument-type]
```

## Actual

```
error[invalid-argument-type]: Argument to function `make_id` is incorrect
  --> repro.py:10:13
   |
10 | x = make_id(UserId, 42)
   |             ^^^^^^ Expected `type[Unknown]`, found `<NewType pseudo-class 'UserId'>`
```

## Expected

No error. A `NewType` pseudo-class is callable with the same signature as `type[T]` and should satisfy `type[T]` parameters.

mypy and pyright both accept this, inferring `T = UserId`.

## Version

```
ty 0.0.16 (e96ea4cd9 2026-02-10)
```

**Comment by AlexWaygood** [comment]
Thanks for the report!

You say that mypy and pyright both accept this, but pyright does not: https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByApgO4AqiRANFBQkQGoCGIAUKwKoDORIAkgBMoAXkKk6RABQAibr0HSaqGAEp2ZEbUrMQMstLWsBRYFAhMA1kQD6SAZIDGAGy4AuWJQDaZALo0AbkxOAK5E7soqUAC0AHy0rqxQSVAgRDDBIChQzlySgSFEhgAemuZWtvZy-AI0ACwATJFQAMRQvOAgnqj5dlEsaMEQRCgwUfD0PkA. And I think mypy is incorrect to do so.

A `type[]` type should be a class or (possibly, though this is debated) a generic alias. The typing conformance suite explicitly asserts that type checkers should understand that a call to `NewType()` does not create an instance of `type` at runtime (it creates a _pseudo_-class, not a real class): https://github.com/python/typing/blob/f98d75ebb241d6f49e1a985a4f259f46b2056626/conformance/tests/aliases_newtype.py#L18

Consider using a `Callable` type instead. This is accepted by ty, and should be by other type checkers also:

```py
from typing import NewType, TypeVar, Callable

UserId = NewType("UserId", int)

T = TypeVar("T")

def make_id(cls: Callable[[int], T], value: int) -> T:
    return cls(value)

x = make_id(UserId, 42)
```
